src/pal/src/include/pal/context.h

```diff
@@ -277,6 +277,25 @@ inline void *FPREG_Xstate_Ymmh(const ucontext_t *uc)
 #define MCREG_Sp(mc)      ((mc).sp)
 #define MCREG_Pc(mc)      ((mc).pc)
 #define MCREG_Cpsr(mc)    ((mc).pstate)
+
+
+inline
+fpsimd_context* GetNativeSigSimdContext(native_context_t *mc)
+{
+    fpsimd_context* fp = reinterpret_cast<fpsimd_context *>(&mc->uc_mcontext.__reserved[0]);
```

Comment:
Hmm. the header says that there's no guarantee for it to be the first. Should we really rely on this?

Reply:
This was how I interterpreted @janvorli suggestion to just assert.

We could add more complexity to support arbitrary order, but since the kernel always puts this first the additional complexity would never be tested.

This is much simpler and cleaner. Aesthetically I like this code better.

@janvorli Any opinions.

Reply:
Interestingly enough, my first thought was exactly what you did here. Then I've thought that being resilient against the fact that it could theoretically be in an arbitrary order would make it more future proof. So my suggestion was about asserting in case we don't find the fpsimd_context it at all, but still have the loop there.
So I would actually prefer that, especially since the header says it can be in an arbitrary order.

Reply:
Done